plugin/forward/health_test.go

```diff
@@ -277,3 +277,24 @@ func TestHealthDomain(t *testing.T) {
 		t.Errorf("Expected number of health checks with Domain==%s to be %d, got %d", hcDomain, 1, i1)
 	}
 }
+
+func TestHealthAllUpstreamsDownServeFail(t *testing.T) {
+	defaultTimeout = 20 * time.Second
+	// Setting bad address for healthcheck to fail
+	p := proxy.NewProxy("TestHealthAllUpstreamsDown", "8.8.8.8:5003", transport.DNS)
+	p1 := proxy.NewProxy("TestHealthAllUpstreamsDown", "8.8.8.8:5004", transport.DNS)
+	f := New()
+	f.SetProxy(p)
+	f.SetProxy(p1)
+	f.hcOnFailAction = "servfail"
+
+	defer f.OnShutdown()
+
+	req := new(dns.Msg)
+	req.SetQuestion("example.org.", dns.TypeA)
+
+	resp, _ := f.ServeDNS(context.TODO(), &test.ResponseWriter{}, req)
+	if resp != dns.RcodeServerFailure {
```

Comment:
this comment somehow missed in my previous review..
With the condition above, you cannot check if your new code works properly. It returns SERVFAIL even if you retry when all proxies are died.

Reply:
Agreed. I updated the test to check for the error message "all upstreams down" but its not working for looking for loopback addresses but it does not work for 8.8.8.8 addresses either.

<img width="1569" alt="image" src="https://github.com/user-attachments/assets/e6976dd1-e9ce-44cd-8c07-1ef7b83a7cfc" />
<img width="725" alt="image" src="https://github.com/user-attachments/assets/39e583b5-dd6c-4c4b-a12c-4c4f319f14f7" />

Reply:
Fixed this, there was a way to mimic the proxy server behavior in other tests. I am checking for servfail and `all upstreams down` message.